services/KakaoService.js

```diff
@@ -0,0 +1,53 @@
+import { KakaoDao } from '../models';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+
+const { JWT_SECRET_KEY } = process.env;
+
+const getKakaoEmail = async (accessToken) => {
+  try {
+    const kakaoUserInfo = await axios({
+      method: 'POST',
+      url: 'https://kapi.kakao.com/v2/user/me',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+      withCredentials: true,
+    });
+
+    const kakaoEmail = kakaoUserInfo.data.kakao_account.email
+      ? kakaoUserInfo.data.kakao_account.email
+      : null;
+
+    return kakaoEmail;
+  } catch (err) {
+    console.log(err);
+    throw new Error(err);
+  }
+};
+
+const createToken = async (kakaoUserInfo) => {
+  const token = jwt.sign({ kakaoUserInfo }, JWT_SECRET_KEY, {
+    expiresIn: '1h',
+  });
+  return token;
+};
+
+const kakaoLogin = async (kakaoEmail) => {
+  try {
+    //kakaoEmail 있는지 다오에서 확인함
+    const existingKakaoUser = await KakaoDao.findUserByEmail(kakaoEmail);
+    //만약에 있으면 바로 토큰 만들어서 반환하고 아니면 다오가서 유저 만들고 다시 토큰 반환
+    if (existingKakaoUser) {
+      return await createToken(existingKakaoUser.email);
+    } else {
+      await KakaoDao.createKakaoUser(kakaoEmail);
+      return await createToken(existingKakaoUser.email);
+    }
```

Comment:
```suggestion
  try {
    //kakaoEmail 있는지 다오에서 확인함
    const existingKakaoUser = await KakaoDao.findUserByEmail(kakaoEmail);
    //만약에 있으면 바로 토큰 만들어서 반환하고 아니면 다오가서 유저 만들고 다시 토큰 반환
    if (existingKakaoUser) {
      return await createToken(existingKakaoUser.email);
    } 
    await KakaoDao.createKakaoUser(kakaoEmail);
    return await createToken(existingKakaoUser.email);
```

참고를 위한 주석도 제거해주세요 :)
위 `if` 문 내부에 return이 있으므로 여기 `else` 없어도 될 것 같습니다 👍